packages/extension/src/contexts/NetworkContext/NetworkContext.tsx

```diff
@@ -1,116 +1,95 @@
-      const network = loadNetwork();
-      const ws = new Client(network.server);
-      try {
-        await ws.connect();
-        setNetwork(network.name);
-        setClient(ws);
-      } catch (err) {
-        await ws?.disconnect();
-        setClient(null);
-        Sentry.captureException(err);
-      }
-    };
-
-    connectToNetwork();
+  const disconnect = useCallback(async () => {
+    if (client && client.isConnected()) {
+      await client.disconnect();
+    }
+    setClient(null);
+    setIsDisconnected(true);
   }, []);
 
-  useEffect(() => {
-    const handleBeforeUnload = () => {
-      // Close the WebSocket connection if it is open
-      if (client) {
-        client?.disconnect();
-      }
-    };
-    // Add event listener for beforeunload event
-    window.addEventListener('beforeunload', handleBeforeUnload);
-
-    // Remove event listener when the component unmounts
-    return () => {
-      window.removeEventListener('beforeunload', handleBeforeUnload);
-    };
-  }, [client]);
-
-  const reconnectToNetwork = async () => {
-    try {
-      const loadedNetwork = loadNetwork();
-      const ws = new Client(network || loadedNetwork.server);
-      await ws.connect();
-      setNetwork(network || loadedNetwork.name);
-      setClient(ws);
-    } catch (err) {
-      await client?.disconnect();
-      setClient(null);
-      Sentry.captureException(err);
-    }
-  };
+  const connect = useCallback(async (loadedNetwork: NetworkInfo) => {
+    const ws = new Client(loadedNetwork.server);
+    await ws.connect();
+    setNetwork(loadedNetwork.name);
+    setClient(ws);
+    client?.on('disconnected', () => {
+      setIsDisconnected(true);
+    });
+    client?.on('connected', () => {
+      setIsDisconnected(false);
+    });
+    setIsDisconnected(false);
+  }, []);
 
-  const switchNetwork = useCallback(
-    async (network: Network) => {
+  const connectToNetwork = useCallback(
+    async (network?: Network) => {
       try {
-        await client?.disconnect();
-        const ws = new Client(NETWORK[network].server);
-        await ws.connect();
-        setNetwork(network);
-        saveNetwork(network);
-        setClient(ws);
+        await disconnect();
+        const loadedNetwork = loadNetwork(network);
+        await connect(loadedNetwork);
+        if (network) saveNetwork(network);
       } catch (err) {
-        await client?.disconnect();
-        setClient(null);
+        await disconnect();
         Sentry.captureException(err);
       }
     },
-    [client]
+    [disconnect, connect]
   );
 
   // Remove Network configuration and set default one
   const resetNetwork = useCallback(async () => {
     try {
       await removeNetwork();
-      const network = await loadNetwork();
-      setNetwork(network.name);
+      await connectToNetwork();
     } catch (err) {
+      await disconnect();
       Sentry.captureException(err);
     }
-  }, []);
+  }, [connectToNetwork, disconnect]);
+
+  useEffect(() => {
```

Comment:
It's better to put the useEffect before all the functions you defines (useCallback) as well it will fits the coding style of the repository.

Reply:
ok